tests/integration/test_pip_packages.py

```diff
@@ -0,0 +1,116 @@
+from requests import get
+import tarfile
+
+from utils import Client
+
+
+def test_pip_package_without_deps(test_env, tmpdir):
+    """
+    Validate data in the pip package request without dependencies.
+
+    Process:
+    Send new request to the Cachito API
+    Send request to check status of existing request
+
+    Checks:
+    * Check that the request completes successfully
+    * Check that a single pip package is identified in response
+    * Check that there are no dependencies in response
+    * Check that the source tarball includes the application source code
+    * Check that the source tarball includes an empty deps/pip directory
+    * TODO: The content manifest is successfully generated and contains correct content
+    """
+    client = Client(test_env["api_url"], test_env["api_auth_type"], test_env.get("timeout"))
+    repo = test_env["pip_packages"]["without_deps"]["repo"]
+    ref = test_env["pip_packages"]["without_deps"]["ref"]
+    pkg_managers = ["pip"]
+    initial_response = client.create_new_request(
+        payload={
+            "repo": repo,
+            "ref": ref,
+            "pkg_managers": pkg_managers,
+            # TODO: delete pip-dev-preview flag when
+            #  the pip package manager will be ready for production usage
+            "flags": ["pip-dev-preview"],
+        },
+    )
+    completed_response = client.wait_for_complete_request(initial_response)
+    assert completed_response.status == 200
+
+    assert_packages_from_response(completed_response.data, 1, pkg_managers)
+
+    assert_dependencies_from_cachito_request(completed_response.data)
+
+    # Download expected source files
+    expected_files_urls = test_env["pip_packages"]["without_deps"]["expected_files"]
+    expected_source_files = {}
+    for file_name, file_url in expected_files_urls.items():
+        expected_source_files[file_name] = get(file_url).text
+
+    # Download and extract source tarball
+    source_name = tmpdir.join(f"download_{str(completed_response.id)}")
+    file_name_tar = tmpdir.join(f"download_{str(completed_response.id)}.tar.gz")
+    resp = client.download_bundle(completed_response.id, file_name_tar)
+    assert resp.status == 200
+    assert tarfile.is_tarfile(file_name_tar)
+    with tarfile.open(file_name_tar, "r:gz") as tar:
+        tar.extractall(source_name)
+
+    # Check that the source tarball includes the application source code under the app directory.
+    assert_expected_files(path.join(source_name, "app"), expected_source_files)
+    # Check that the source tarball includes an empty deps directory.
+    assert_expected_files(path.join(source_name, "deps"))
+
+
+def assert_packages_from_response(response_data, pkg_count, pkg_managers):
+    """Check amount and type of packages in the response data."""
+    assert len(response_data["packages"]) == pkg_count
+    for idx in range(pkg_count):
+        assert response_data["packages"][idx]["type"] == pkg_managers[idx]
```

Comment:
Is this assertion correct? In other words, would Cachito require you to repeat the same package manager multiple times in the request in case it would handle a request for multiple packages?

Reply:
I am not testing requirements from cachito. I am testing  `packages/type` from: 
```
  "packages": [
    {
      "dependencies": [], 
      "name": "cachito-pip-empty", 
      "type": "pip", 
      "version": "1.0.0"
    }
  ], 
  "pkg_managers": [
    "pip"
  ], 
```
Maybe I got you in the wrong way. Should I change `pkg_managers` to `pkg_manager`?

Reply:
the issue here is that if you have multiple packages in `response_data["packages"]` (Cachito does support this), you will need to repeat yourself when declaring `pkg_managers` at https://github.com/release-engineering/cachito/pull/270/files#diff-984b9acc3c13f3b64caac70e2a065e1eR29. I am not sure if doing so is even supported by Cachito.

Reply:
Me too right now. I thought I will be able to identify and change it with multiple package test. I just separated it into a few PRs to be able to provide basic assertions. I am not sure what is the best way to do it right now. @chmeliik What do you think?